### PR TITLE
Fr/feature/documentation sidebar

### PR DESCRIPTION
**Pull request by RavenInd**
Create sidebar and mobile sidebar

### CODE REVIEW

**Review by usulpro — CHANGES_REQUESTED** [review]
#### This section jumps to right when selecting a subitem:

image

#### Is there a reason to switch to mobile view under 1440px? It still desktop and there is a lot of space
image


#### Fonts looks different, top padding is different
image

#### Menu item area

Compare it with original. It's important to keep clickable area as much as possible, especially for mobile users

prod:

image

ours:

image

#### In the mobile version sidebar doesn't close after selecting another item

but ask Patricio what behavior is preferable

#### Mobile sidebar isn't closing by clicking outside
image

#### Mobile header is scrollable itself, while shouldn't
image

#### Fonts, dividers brightness, missed scrollbar
image

#### Bottom padding
image

File: docs/src/themes.tsx [hunk 1]
```diff
@@ -25,9 +25,11 @@ export const appTheme = {
     neutral500: "#4B4D4D",
     neutral600: "#6E6F70",
     neutral700: "#9B9FA8",
+    neutral800: "#16181D",
     neutral900: "#0A0A0A",
     accent100: "#FBFCDB",
     accent600: "#FFF04D",
+    accent700: "#CCB200",
```
Comment (usulpro) [inline docs/src/themes.tsx:34]: Do I missing something but colors from 600 to 900 don't look like they gradually ranged

File: docs/src/config.ts [hunk 2]
```diff
@@ -63,3 +63,224 @@ export const SOCIALS_LINKS = {
 
 export const BANNER_LINK =
   "https://www.notion.so/Nomic-Foundation-jobs-991b37c547554f75b89a95f437fd5056";
+
+export const DocumentationSidebarStructure = [
```
Comment (usulpro) [inline docs/src/config.ts]: TODO: move it to getStaticProps

File: docs/src/components/DocumentationLayout.tsx [hunk 3]
```diff
@@ -76,11 +125,25 @@ const DocumentationLayout = ({ children, seo }: Props) => {
           <DefaultBanner content={content} />
         )}
       />
-      <Navigation />
+      <Navigation
+        isSidebarOpen={isSidebarOpen}
+        setIsSidebarOpen={setIsSidebarOpen}
```
Comment (usulpro) [inline docs/src/components/DocumentationLayout.tsx]:
```suggest
onSidebarOpen={setIsSidebarOpen}
```

let's have a convention to name all callback props same style

File: docs/src/components/Sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+
+import { styled } from "linaria/react";
+import { tm } from "../themes";
+import { DocumentationSidebarStructure } from "../config";
+
+interface Props {
+  elementsList: typeof DocumentationSidebarStructure;
```
Comment (usulpro) [inline docs/src/components/Sidebar.tsx]: Let's specify this type explicitly. The reason - we will generate this structure at building time so we won't have this shape in config

File: docs/src/components/Sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+              </SidebarLinkWrapper>
+            ) : (
+              <SidebarHeading>{sidebarItem.label}</SidebarHeading>
+            )}
+
+            {sidebarItem?.children && (
+              <SidebarSubLinksList>
+                {sidebarItem.children.map((subItem) => {
+                  const isSubLinkActive = router?.asPath === subItem.href;
+                  const isAnchor = router?.asPath.indexOf("#") > -1;
```
Comment (usulpro) [inline docs/src/components/Sidebar.tsx]: router?.asPath.includes("#") - doesn't this work?

**Review by usulpro — CHANGES_REQUESTED** [review]

File: docs/src/themes.tsx (hunk 1 above)
Comment (RavenInd) [reply docs/src/themes.tsx:34]: Will do it after all colours will be add

File: docs/src/components/DocumentationLayout.tsx (hunk 3 above)
Comment (RavenInd) [reply docs/src/components/DocumentationLayout.tsx]: corrected

File: docs/src/components/Sidebar.tsx (hunk 4 above)
Comment (RavenInd) [reply docs/src/components/Sidebar.tsx]: Done

File: docs/src/components/Sidebar.tsx (hunk 5 above)
Comment (RavenInd) [reply docs/src/components/Sidebar.tsx]: Refactored

File: docs/src/config.ts (hunk 2 above)
Comment (RavenInd) [reply docs/src/config.ts]: ok

**Pull request merged**